### PR TITLE
Changes for EOS-17931 CORTX-HA : Get support bundle working in R2

### PR DESCRIPTION
**Pull request by ArchanaLimaye**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    Support bundle creation needs to be added 
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
   Support bundle creation needs to be added 
  </code>
</pre>
## Solution
<pre>
  <code>
    Support bundle creation added to the new CLI framework 
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Confirmed that support bundle gets created with CLI for the same is issued 
    Confirmed that support bundle help is displayed when "-h" is issued for the same 
  </code>
</pre>

### CODE REVIEW

File: ha/cli/exec/supportBundleExecutor.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        """
+        super(CreateSupportBundleExecutor, self).__init__()
+
+    def _add_more_log(self):
+        """
+        Collect HA commands Log
+        """
+        pass
+
+    def _check_files(self, ha_log_list=const.CORTX_SUPPORT_BUNDLE_LOGS):
```
Comment (ajay-paratmandali) [inline ha/cli/exec/supportBundleExecutor.py:143]: Modify
```
def _check_files(self, ha_log_list=None):
          if ha_log_list == None:
                 ha_log_list = const.CORTX_SUPPORT_BUNDLE_LOGS
```

Comment (azheregelya) [reply ha/cli/exec/supportBundleExecutor.py]: @ajay-paratmandali what's the trick? Look equal.

File: ha/cli/exec/supportBundleExecutor.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        '''
+        parser = argparse.ArgumentParser(prog='support_bundle create')
+        parser.add_argument("support_bundle", help="Module name")
+        parser.add_argument("create", help="Operation")
+        parser.add_argument('--id', help='Bundle id', required=True)
+        parser.add_argument('--path', help='Bundle path', required=True)
+        self._args = parser.parse_args()
+        self._id = self._args.id
+        self._path = self._args.path
+        return True
```
Comment (azheregelya) [inline ha/cli/exec/supportBundleExecutor.py:54]: `return True` looks redundant. Is it required by some interface?

File: ha/cli/exec/supportBundleExecutor.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+
+    def validate(self) -> str:
+        self.parse_cluster_args()
+        return True
```
Comment (azheregelya) [inline ha/cli/exec/supportBundleExecutor.py:59]: Incompatible return type.

File: ha/const.py [hunk 4]
```diff
@@ -25,9 +25,10 @@
 PCSD_LOG="/var/log/pcsd/pcsd.log"
 HA_CMDS_OUTPUT="{}/ha_cmds_output".format(RA_LOG_DIR)
 COROSYNC_LOG="/var/log/cluster"
+ETC_HA_DIR="/etc/cortx/ha/"
```
Comment (azheregelya) [inline ha/const.py:28]: I believe this path is already defined by CONFIG_DIR

**Review by azheregelya — CHANGES_REQUESTED** [review]

File: ha/cli/command_factory.py [hunk 5]
```diff
@@ -73,6 +73,11 @@ def __init__(self):
                 "-h": "ha.cli.exec.commandExecutor.ServiceCLIUsage",
                 "--help": "ha.cli.exec.commandExecutor.ServiceCLIUsage"
             },
+            "support_bundle": {
```
Comment (indrajitzagade) [inline ha/cli/command_factory.py:76]: Please comment on if it was decided to have this as a CLI.

File: ha/cli/exec/supportBundleExecutor.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+    def _add_more_log(self):
+        """
+        Collect Extra Log
+        """
+        cmd_list = [const.PCS_STATUS, const.PCS_FAILCOUNT_STATUS]
+        with open(const.HA_CMDS_OUTPUT, "w") as file:
+            for cmd in cmd_list:
+                _err = ""
+                _proc = SimpleProcess(cmd)
```
Comment (indrajitzagade) [inline ha/cli/exec/supportBundleExecutor.py:92]: I think we use SimpleCommand()

File: ha/cli/command_factory.py (hunk 5 above)
Comment (ArchanaLimaye) [reply ha/cli/command_factory.py:76]: This is a pre-existing CLI from 1.0 
I have ensured that the same is present here

File: ha/cli/exec/supportBundleExecutor.py (hunk 6 above)
Comment (ArchanaLimaye) [reply ha/cli/exec/supportBundleExecutor.py:92]: I have retained the pre-existing support bundle code as is. 
Maybe we can take up changes in the enhancement ticket EOS-17935 [CORTX-HA : Redesign support bundle generation] ??

File: ha/const.py (hunk 4 above)
Comment (ArchanaLimaye) [reply ha/const.py]: Removed the duplicate path

File: ha/cli/exec/supportBundleExecutor.py (hunk 3 above)
Comment (ArchanaLimaye) [reply ha/cli/exec/supportBundleExecutor.py:59]: Modified return type

File: ha/cli/exec/supportBundleExecutor.py (hunk 2 above)
Comment (ArchanaLimaye) [reply ha/cli/exec/supportBundleExecutor.py:54]: Return value checks added to the caller

File: ha/cli/command_factory.py (hunk 5 above)
Comment (indrajitzagade) [reply ha/cli/command_factory.py:76]: @ajaysrivas So this will be CLI, is it?

File: ha/cli/exec/supportBundleExecutor.py (hunk 6 above)
Comment (indrajitzagade) [reply ha/cli/exec/supportBundleExecutor.py:93]: ok

File: ha/cli/command_factory.py (hunk 5 above)
Comment (indrajitzagade) [reply ha/cli/command_factory.py:76]: Fine with the CLI.

**Review by azheregelya — APPROVED** [review]

File: ha/cli/exec/supportBundleExecutor.py (hunk 1 above)
Comment (ajay-paratmandali) [reply ha/cli/exec/supportBundleExecutor.py]:
```
    def _check_files(self, ha_log_list=const.CORTX_SUPPORT_BUNDLE_LOGS):
```
It provide some codacy issue to use constant variable assign in parameter 
To address codacy issue we can change format 
```
ha_log_list=const.CORTX_SUPPORT_BUNDLE_LOGS
```
Here we can assign string or none but not variable as per codacy rule

**Review by ajay-paratmandali — APPROVED** [review]

**Review by indrajitzagade — APPROVED** [review]

File: conf/etc/v2/cli_schema.json [hunk 7]
```diff
@@ -116,8 +116,8 @@
     },
     "support_bundle": {
         "create": {
-            "usage": "",
-            "class": ""
+            "usage": "support_bundle create  <Bundle ID> <Bundle Path>",
```
Comment (indrajitzagade) [inline conf/etc/v2/cli_schema.json:119]: Is the usage string correct now?

Comment (ArchanaLimaye) [reply conf/etc/v2/cli_schema.json:119]: Yes its exactly same as V1 now

**Review by ajay-paratmandali — APPROVED** [review]

**Pull request merged**